### PR TITLE
Add license check for Unicode/ICU License

### PR DESCRIPTION
**Pull request by nikolag-ikor**
# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing (regression) unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Hauke-Schroeder** [comment]
@nikolag-ikor we are missing the license file in this folder: https://github.com/IKOR-GmbH/sip-framework/tree/develop/licenses

File: .github/workflows/check-licenses.sh [hunk 1]
```diff
@@ -44,6 +44,7 @@ while read line; do
       [[ $line == *"(The MIT License)"* ]] ||
       [[ $line == *"(MIT License)"* ]] ||
       [[ $line == *"(MIT)"* ]] ||
+      [[ $line == *"(Unicode/ICU License)"* ]] ||
```
Comment (MartinBuchheim) [inline .github/workflows/check-licenses.sh:47]: Did we get confirmation that it's ok to use that license?

**Review by MartinBuchheim — COMMENTED** [review]
Hey Nikola. Two points:
* Please confirm that we have the OK to use ICU license (see my other remark)
* This PR is currently based on `develop`, not `develop-v3` . Is that on purpose?

File: .github/workflows/check-licenses.sh (hunk 1 above)
Comment (nikolag-ikor) [reply .github/workflows/check-licenses.sh:47]: Hi Martin,

Not yet, Hauke asked me to prepare PR and he will resolve it once when he gets the confirmation for the license.
He also proposed to make the PR on develop branch.

**Review by Hauke-Schroeder — APPROVED** [review]

**Pull request merged**